mesonbuild/backend/ninjabackend.py

```diff
@@ -96,39 +110,68 @@ def __init__(self, rule, command, args, description,
         self.extra = extra
         self.rspable = rspable  # if a rspfile can be used
         self.refcount = 0
+        self.rsprefcount = 0
 
     def write(self, outfile):
-        if not self.refcount:
-            return
+        for rsp in ([''] if self.refcount else [] +
+                    ['_RSP'] if self.rsprefcount else []):
+            outfile.write('rule {}{}\n'.format(self.name, rsp))
+            if self.rsprefcount:
```

Comment:
Surely this should by `if rsp.endswith('_RSP'):`?  Now I'm wondering how this ever worked :)

Reply:
dag.  good catch.